e2e/deployers/retry.go

```diff
@@ -68,3 +76,88 @@ func WaitWorkloadHealth(ctx types.TestContext, cluster types.Cluster, namespace
 		}
 	}
 }
+
+func WaitForResourceDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	key k8stypes.NamespacedName,
+	obj client.Object,
+) error {
+	log := ctx.Logger()
+	resourceType := getResourceTypeName(obj)
+	startTime := time.Now()
+
+	log.Debugf("Waiting until %s \"%s/%s\" is deleted in cluster %q",
+		resourceType, key.Namespace, key.Name, cluster.Name)
+
+	for {
+		if err := cluster.Client.Get(ctx.Context(), key, obj); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				return err
+			}
+
+			log.Debugf("%s \"%s/%s\" deleted in cluster %q",
+				resourceType, key.Namespace, key.Name, cluster.Name)
+
+			return nil
+		}
+
+		if time.Since(startTime) > 60*time.Second {
+			return fmt.Errorf("timeout deleting %s \"%s/%s\" in cluster %q",
+				resourceType, key.Namespace, key.Name, cluster.Name)
+		}
+
+		if err := util.Sleep(ctx.Context(), time.Second); err != nil {
+			return err
+		}
+	}
+}
+
+func WaitApplicationSetDeleted(ctx types.TestContext) error {
+	return WaitForResourceDelete(
+		ctx,
+		ctx.Env().Hub,
+		k8stypes.NamespacedName{
+			Name:      ctx.Name(),
+			Namespace: ctx.ManagementNamespace(),
+		},
+		&argocdv1alpha1hack.ApplicationSet{},
```

Comment:
Nice!

Reply:
We can pass the name and the namespace using:

    &argocdv1alpha1hack.ApplicationSet{
        ObjectMeta: types.ObjectMeta{
            Name: name,
            Namespace: namespace,
        }
    },

The advantage of packing all the info in the object is being able to wait for multiple objects in the same cluster with a small change.